app/jobs/dqt/sync_teachers_job.rb

```diff
@@ -7,7 +7,9 @@ class SyncTeachersJob < ApplicationJob
     def perform(batch_size = 500, interval = 30.seconds)
       return unless FeatureService.enabled?("dqt_import.sync_teachers")
 
-      trainees = Trainee.where.not(trn: nil).where("length(trn) = 7").in_training.or(Trainee.deferred)
+      trainees = Trainee.where.not(trn: nil).where("length(trn) = 7").in_training.or(
```

Comment:
is it worth extracting this to a scope on `Trainee`? or some other way of making it clear what the `.where("length(trn) = 7")` is all about.

Reply:
Yeah it's a hidden detail. I think DQT will reject any update request that doesn't have a 7 char length trn. I wonder whether to just include it in the existing `in_training` and `deferred` scopes 🤔

Reply:
That sounds sensible. Is there any situation where a trn length would be something other than 7? Could be a case for postgres-level validation on that field.

Reply:
as far as I'm aware it's always been a 7 char length even from DTTP days (I think) so not sure why we're doing the extra check (it was in an earlier data migration so I assumed something may have changed to require this but honestly don't know)

Reply:
if the rationale is for avoiding updates in non-prod environments, probably we should think of a better way of doing this. or at least being explicit about it.